llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -120,6 +120,34 @@ bool AMDGPUInstructionSelector::constrainCopyLikeIntrin(MachineInstr &MI,
          RBI.constrainGenericRegister(Src.getReg(), *SrcRC, *MRI);
 }
 
+// Returns true if this is a copy from physical reg to VCC.
+// The parameter IsValid further indicates if the copy is valid.
+bool AMDGPUInstructionSelector::isCopyPhysicalToVCC(
+    Register DstReg, Register SrcReg, const MachineRegisterInfo &MRI,
+    bool &IsValid) const {
+  if (DstReg.isPhysical())
+    return false;
+  if (!SrcReg.isPhysical() || SrcReg == AMDGPU::SCC)
+    return false;
+
+  auto &RegClassOrBank = MRI.getRegClassOrRegBank(DstReg);
+  const TargetRegisterClass *RC =
+      RegClassOrBank.dyn_cast<const TargetRegisterClass *>();
+  if (RC) {
+    const LLT Ty = MRI.getType(DstReg);
+    if (!Ty.isValid() || Ty.getSizeInBits() != 1)
+      return false;
+    // G_TRUNC s1 result is never vcc.
+    if (MRI.getVRegDef(DstReg)->getOpcode() == AMDGPU::G_TRUNC ||
+        !RC->hasSuperClassEq(TRI.getBoolRC()))
+      return false;
+    IsValid = RC->contains(SrcReg);
```

Comment:
You do not need to handle invalid cases. Invalid cases should fail the verifier

Reply:
> You do not need to handle invalid cases. Invalid cases should fail the verifier

If we can assume that invalid cases won't reach `selectCOPY()`, then the code is simplified to just 2 lines. I don't think we need a separate function, because that would just be moving code from `selectCOPY()` to another function.